wurst/math/Raycast.wurst

```diff
@@ -0,0 +1,483 @@
+package Raycast
+import NoWurst
+import Vectors
+import Wurstunit
+
+/* The package defines such geometry as 'ray' for 2D and 3D separately.
+Both can find the intersection point with another geometry object.
+
+Most 2D functions return result2d(success, point, distance) tuple which contains
+information about intersection where
+'success' is 'true' if the ray intersects the given object
+'point' is the intersection point or zero-vector
+'distance' is the distance between the ray's origin and the intersection point
+
+A ray can intersect a convex object (e.g. a circle) zero, one or two times at once.
+In this case the function returns doubleresult2d tuple that contains a couple of result2d
+tuples.
+
+Functions that find 3D intersection work similarly and return result3d tuple.
+result3d(success, point, distance, backface) also contains 'backface' boolean value that's
+true in case the object is faced away from  the ray's origin. Functions which find the
+intersection point with a convex object (e.g. a sphere) return doubleresult3d tuple.
+
+Also you can project a vector onto a ray to obtain the point on the ray that's nearest
+to the vector.
+
+Interacting beams, projectile collision, protective fields and many more can be created
+via raycast. */
+
+constant FAIL_2D = result2d(false, ZERO2, 0)
+constant FAIL_3D = result3d(false, ZERO3, 0, false)
+constant EPSILON = 0.0001
+
+/** An infinite 2D ray
+origin - the point which the ray comes from
+direction - directional vector */
+public tuple ray2d(vec2 origin, vec2 direction)
+
+/** An infinite 3D ray
+origin - the point which the ray comes from
+direction - directional vector */
+public tuple ray3d(vec3 origin, vec3 direction)
+
+/** 2D intersection result
+success - 'true' if the intersection point exists
+point - the intersection point or ZERO2
+distance - distance between the ray's origin and the intersection point */
+public tuple result2d(bool success, vec2 point, real distance)
+
+/** Double 2D intersection result.
+'first' is nearest to the ray's origin. */
+public tuple doubleresult2d(result2d first, result2d second)
+
+/** 3D intersection result
+success - 'true' if the intersection point exists
+point - the intersection point or ZERO2
+distance - distance between the ray's origin and the intersection point 
+backface - `true` if the surface at the intersection point is faced away from the ray's origin  */
+public tuple result3d(bool success, vec3 point, real distance, bool backface)
+
+/** Double 3D intersection result.
+'first' is nearest to the ray's origin. */
+public tuple doubleresult3d(result3d first, result3d second)
+
+/* =============== 2D =============== */
+
+function vec2.cross(vec2 v) returns real
```

Comment:
why is this here? Belongs into vector

Reply:
Because there's no cross product of two 2d vectors. This is a simplification for several functions within the package.